app/models/chargeback/consumption_with_rollups.rb

```diff
@@ -0,0 +1,44 @@
+class Chargeback
+  class ConsumptionWithRollups < Consumption
+    delegate :timestamp, :resource, :resource_id, :resource_name, :resource_type, :parent_ems,
+             :hash_features_affecting_rate, :tag_list_with_prefix, :parents_determining_rate,
+             :to => :first_metric_rollup_record
+
+    def initialize(metric_rollup_records, start_time, end_time)
+      super(start_time, end_time)
+      @rollups = metric_rollup_records
+    end
+
+    def tag_names
+      first_metric_rollup_record.tag_names.split('|')
+    end
+
+    def max(metric)
+      values(metric).max
```

Comment:
I don't think this overlaps directly. This is already optimized code compared to what we have in euwe or even before. Feel free to take a look.

Previously, we went one metric_rollup per cycle. Now, we take them in batches. So, it is not really moved code. I created this before. If you go pr by pr into history of CB you will see how performance improved in every step.

I don't think we can re-use this code from other report as we have very custom requirements for chargebacks. Although, I'll be interested to learn where lays any similar code.